### PR TITLE
Sprint22 microstructure

### PR DESCRIPTION
**Pull request by mkuehbach**

### CODE REVIEW

File: contributed_definitions/nyaml/NXcoordinate_system_set.yaml [hunk 1]
```diff
@@ -88,138 +88,259 @@ doc: |
   always guide the user for each group and field which of the coordinate system
   one refers to.
 type: group
+# express conventions explicitly and understandable
+# appdefs each instance of NXentry should have its own set of conventions
+# method-specific additional conventions on top of the ones formulated here
+# should go to method-specific base classes like NXcoordinate_system_em_ebsd
+# use depends_on field - not attribute - to point to conventions which in specific cases
 NXcoordinate_system_set(NXobject):
+  rotation_handedness(NX_CHAR):
```
Comment (lukaspie) [inline contributed_definitions/nyaml/NXcoordinate_system_set.yaml:97]: would that not be someting that is defined per coordinate system. Or do you want to say, we use the same convention for all CSs?

File: contributed_definitions/nyaml/NXcoordinate_system_set.yaml [hunk 2]
```diff
@@ -88,138 +88,259 @@ doc: |
...
+      DOI: 10.1088/0965-0393/23/8/083501.
+    enumeration: [undefined, passive, active]
+  euler_angle_convention(NX_CHAR):
+    doc: |
+      How are Euler angles interpreted given that there are several
+      choices (e.g. ZXZ, XYZ, etc.) according to convention 4 of
+      DOI: 10.1088/0965-0393/23/8/083501.
+      The most frequently used convention is ZXZ which is based on
+      the work of H.-J. Bunge but other conventions are possible.
+    enumeration: [undefined, zxz]
```
Comment (lukaspie) [inline contributed_definitions/nyaml/NXcoordinate_system_set.yaml:119]: if there are several choices (you even mention some above), they should be reflected here.

File: contributed_definitions/nyaml/NXcoordinate_system_set.yaml [hunk 3]
```diff
@@ -88,138 +88,259 @@ doc: |
...
+      Details about eventually relevant named directions that may
+      give reasons for anisotropies. The classical example is cold-rolling
+      where one has to specify which directions (rolling, transverse, and normal)
+      align how with the direction of the base vectors of the sample_reference_frame.
+    type(NX_CHAR):
+      doc: |
+        Type of coordinate system and reference frame according to
+        convention 1 of DOI: 10.1088/0965-0393/23/8/083501.
+      enumeration: [undefined, cartesian]
+    handedness(NX_CHAR):
```
Comment (lukaspie) [inline contributed_definitions/nyaml/NXcoordinate_system_set.yaml:146]: Many of these fields are repeating what is anyway in the base class `NXcoordinate_system`, so they can be removed even if we want to keep these specific instances of `NXcoordinate_system`.

File: contributed_definitions/nyaml/NXcoordinate_system_set.yaml [hunk 4]
```diff
@@ -88,138 +88,259 @@ doc: |
...
+  sample_reference_frame(NXcoordinate_system):
+    doc: |
+      Details about the sample/specimen reference frame.
+    type(NX_CHAR):
+      doc: |
+        Type of coordinate system and reference frame according to
+        convention 1 of DOI: 10.1088/0965-0393/23/8/083501.
+        The reference frame for the sample surface reference is used for
+        identifying positions on a (virtual) image which is formed by
+        information collected from an electron beam scanning the
```
Comment (lukaspie) [inline contributed_definitions/nyaml/NXcoordinate_system_set.yaml:204]: This is much too specific (to EM) for this base class, this should either be made more general or go to the appdef.

File: contributed_definitions/nyaml/NXcoordinate_system_set.yaml [hunk 5]
```diff
@@ -88,138 +88,259 @@ doc: |
...
-#     <group type="NXcoordinate_system"/>
-#     <!--How to interpret "depends_on" calls for NXtransformations instances which live
-# within an instance of NXcoordinate_system(_set)?
-# Depends on cardinality of NXcoordinate_system_set, i.e. how many NXcoordinate_system are defined?
-# - 0, the root of an NXtransformation chain i.e. "." means McStas
-# - 1, the root of an NXtransformation chain i.e. "." means "that specific CS" i.e. the one defined by the single instance of NXcoordinate_system in the set
-# - > 1, "." becomes ambiguous. My suggestion, spell out the name of the specific NXcoordinate_system instance to be used, using "."
-# despite all these, a warning should be raised, like a logic warning (and McStas kick in) or an error (i.e. be forbidden) be raised-->
-# </definition>
+  # frequently used coordinate systems with frequently relevant specializations
```
Comment (lukaspie) [inline contributed_definitions/nyaml/NXcoordinate_system_set.yaml:134]: I am not a big fan of having these explicitly named coordinate systems here? What do we gain from that? They should be defined on the appdef level.

File: contributed_definitions/nyaml/NXcoordinate_system_set.yaml [hunk 6]
```diff
@@ -88,138 +88,259 @@ doc: |
...
+        Type of coordinate system and reference frame according to
+        convention 1 of DOI: 10.1088/0965-0393/23/8/083501.
+      enumeration: [undefined, cartesian]
+    handedness(NX_CHAR):
+      doc: |
+        Handedness of coordinate system.
+      enumeration: [right_handed, left_handed]
+    origin(NX_CHAR):
+      doc: |
+        Location of the origin of the processing_reference_frame.
```
Comment (lukaspie) [inline contributed_definitions/nyaml/NXcoordinate_system_set.yaml:152]: This sort of text is repeated in all instances of `NXcoordinate_system` here. We can just add it one to the base class and be done.

File: contributed_definitions/nyaml/NXcoordinate_system_set.yaml [hunk 7]
```diff
@@ -88,138 +88,259 @@ doc: |
...
+      enumeration: [undefined, front_top_left, front_top_right, front_bottom_right, front_bottom_left, back_top_left, back_top_right, back_bottom_right, back_bottom_left]
+    x_alias(NX_CHAR):
+      doc: |
+        Name or alias assigned to the x-axis base vector,
+        e.g. rolling direction.
+    x_direction(NX_CHAR):
+      doc: |
+        Direction of the positively pointing x-axis base vector of
+        the processing_reference_frame. We assume the configuration
+        is inspected by looking towards the sample surface from a position
```
Comment (lukaspie) [inline contributed_definitions/nyaml/NXcoordinate_system_set.yaml:168]: Way is that an assumption that should always be taken? The individual communities can have different agreements about their prefered coordinate system.

File: contributed_definitions/nyaml/NXcoordinate_system_set.yaml [hunk 8]
```diff
@@ -88,138 +88,259 @@ doc: |
...
+        also the help info for the xaxis_direction field.
+      enumeration: [undefined, north, east, south, west, in, out]
+  detector_reference_frameID(NXcoordinate_system):
+    doc: |
+      Details about the detector reference frame for a specific detector.
+    \@detector:
+      type: NX_CHAR
+      doc: |
+        Reference to the specifically named :ref:`NXdetector` instance
+        for which these conventions in this :ref:`NXprocess` group apply
```
Comment (lukaspie) [inline contributed_definitions/nyaml/NXcoordinate_system_set.yaml:279]: there is no `NXprocess` group here

File: contributed_definitions/nyaml/NXcoordinate_system_set.yaml [hunk 9]
```diff
@@ -88,138 +88,259 @@ doc: |
...
+        e.g. longest edge as some landmark on the detector.
+    x_direction(NX_CHAR):
+      doc: |
+        Direction of the positively pointing x-axis base vector of
+        the detector space reference frame. We assume the configuration
+        is inspected by looking towards the sample surface from a
+        position that is located behind the detector.
+        Different tools assume that different strategies can be used
+        and are perceived as differently convenient to enter
+        details about coordinate system definitions. In this ELN users
```
Comment (lukaspie) [inline contributed_definitions/nyaml/NXcoordinate_system_set.yaml:313]: There is no ELN here?

File: contributed_definitions/nyaml/NXcoordinate_system_set.yaml [hunk 10]
```diff
@@ -88,138 +88,259 @@ doc: |
...
+      doc: |
+        Direction of the positively pointing z-axis base vector of
+        the detector space reference frame. We assume the configuration
+        is inspected by looking towards the sample surface from a
+        position that is located behind the detector.
+        For further information consult also the help info for the
+        xaxis_direction field.
+      enumeration: [undefined, north, east, south, west, in, out]
+  # conventions specific for certain methods
+  (NXcoordinate_system_em_ebsd):
```
Comment (lukaspie) [inline contributed_definitions/nyaml/NXcoordinate_system_set.yaml:346]: `NXcoordinate_system_em_ebsd` should not be part of the generic `NXcoordinate_system_set`, but rather an instance of `NXcoordinate_system` somewhere in the ebsd description.

**Review by lukaspie — CHANGES_REQUESTED** [review]
Only reviewed changes to `NXcoordinate_system_set` so far.

I support having the conventions liker `rotation_handedness`, `rotation_convention`, `euler_angle_conventions`, and so on in `NXcoordinate_system_set` (though maybe can be different for different coordinate systems, not sure). 

I am of two minds w.r.t. explictily defining different coordinate systems within the base class `NXcoordinate_system_set`. Essentially, this just makes sure all appdefs use the same name if they are talking about a coordinate system of e..g. the sample. This may be helpful. That being said, the coordinate system descriptions here are 1) too specific still (mostly to EM), and 2) oftentimes just repeating what is already in the `NXcoordinate_system` base class, so we can delete most of the fields in these named CSs.

File: contributed_definitions/nyaml/NXcoordinate_system_set.yaml (hunk 1 above)
Comment (mkuehbach) [reply contributed_definitions/nyaml/NXcoordinate_system_set.yaml:97]: I see your arguments and I agree they are substantiated yet I feel a bit like I am running in circles with this:

These five fields are specific concepts based on community suggested best practices how to describe 3D rotations as their docstrings specify, cleaner than most of our docstrings certainly in EM. Therefore, I see a substantiated argument to have them as an own group "say <<that community conventions>>" as this is semantically a category (conventions)

Now comes NeXus and their best practices, here I have two choices:
1a) Add this group of fields to an existent base class
1b) or make this group a own base class
2.) Have that defined on the fly i.e. in an appdef.

My original suggestion was 1b) the view of NeXus or at least our guess of NIAC point will be why to have this as an own base class, we do not wish that the number of base classes increases like crazy, I agree and at the same time I disagree because.
I would like to be able to follow a single source principle that means if I write an appdef
I dont want to repeat all these exact definitions again but just use them, i.e. use that <<conventions>> base class it is clearer it avoids spurious mistakes by copying over docstrings incorrectly and it keeps the appdef cleaner.

I agree that these sort of conventions are related only to the discussion of coordinate system.
Forget the symbols used for these fields for a sec and read only the docstring which is de facto concept. Then it is clear that it is not the handedness of an instance NXcoordinate_system but the handedness for a rotation about an axis an axis that can live in any 3D coordinate system.

An alternative would be to place these definitions to NXrotation_set where these would fill better. But used in practice then each NXrotation_set instance would need to have these definitions spelled out again or have pointer to it e.g. along the lines of a "depends_on" field etc or again some convention hidden in the NeXus docs. I dont want any of these later opinions. I think we can be sportive enough to substantiate why having even small well-defined base classes can be useful.

The true discussion in my opinion always oscillates between two extremes either put concept in base classes which ought not become too packed glossaries nor too domain specific or to put it in an appdef and then for any other appdef you have to start defining it again.

I think I should revert this to solution 1b and have NXcoordinate_system_rotation_convention.yaml

File: contributed_definitions/nyaml/NXcoordinate_system_set.yaml (hunk 2 above)
Comment (mkuehbach) [reply contributed_definitions/nyaml/NXcoordinate_system_set.yaml:119]: Good point, implemented

File: contributed_definitions/nyaml/NXcoordinate_system_set.yaml (hunk 3 above)
Comment (mkuehbach) [reply contributed_definitions/nyaml/NXcoordinate_system_set.yaml:146]: No these are different concepts which have no equally specific docstring counterparts currently in NXcoordinate_system_set, see my first longer point

File: contributed_definitions/nyaml/NXcoordinate_system_set.yaml (hunk 4 above)
Comment (mkuehbach) [reply contributed_definitions/nyaml/NXcoordinate_system_set.yaml:204]: Same argument as already detailed, in this case I think I should revert it to NXcoordinate_system_SOMETHING but neither em nor em_ebsd would be correct because also in sample preparation you could already by virtue of the design of the experiment prepare your specimen physically using these conventions in mind. Unfortunately with area A we hadnt had the discussion neither to such level of detail nor to something fruitful which is why right now there is not even a custom schema in NOMAD for how you have polished your specimen. I agree with you that the docstring is indeed mentioning EM currently explicitly and this is a great comment insofar as really I should have cleaned that while recategorizing that concept (and I guess a few others) but that is the point, the concept here is specifically explaining using references for instances of concepts available in the field where the definition is used and indeed I agree with you that makes a good point to place it into an appdef but then this gets longer and I am sure that NXem will also at some point need to specialize to NXem_SOMETHING_MORE_SPECIFIC and then one would have to copy the concepts over again with exactly that docstring or have an even deeper inheritance chain where NXem_SOMETHING extends NXem ...

I feel it is really tricky to stick always to this idea that we should have an as minimal number of concepts as possible while in several de facto existent OWL-expressed artifacts and instantiated example people do not even follow the idea of a tree.

File: contributed_definitions/nyaml/NXcoordinate_system_set.yaml (hunk 1 above)
Comment (lukaspie) [reply contributed_definitions/nyaml/NXcoordinate_system_set.yaml:97]: As is said in my main statement above, I think keeping these conventions liker `rotation_handedness`, `rotation_convention`, `euler_angle_conventions`, and so on in NXcoordinate_system_set is a good idea. So I would advocate against `NXcoordinate_system_rotation_convention`´and just keep them in `NXcoordinate_system_set`. The main argument is that `NXcoordinate_system_set` is, among others, to be used to describe the community-agreed coordinate system, that it makes sense to bundle these with the community-agreed conventions.

So tldr, keep them here if you agree.

Comment (lukaspie) [reply contributed_definitions/nyaml/NXcoordinate_system_set.yaml:97]: These are also not domain-specific, so they deserve to be in this base class.

File: contributed_definitions/nyaml/NXcoordinate_system_set.yaml (hunk 5 above)
Comment (mkuehbach) [reply contributed_definitions/nyaml/NXcoordinate_system_set.yaml:134]: To be able to used them just as they are defined (see my previous point) avoid creating conceptual inconsistencies during appdef writing when docstrings are not fully copied over etc. etc, like building blocks and the more I think about it again I feel the idea of introducing some bridge between NeXus = set of super generic base classes + { {super specific base classes + specializations + constraints }} i.e. appdef would be useful like sub-parts of concept graphs that a community can reuse which are stronger specialized than base classes but can be reutilized for composing them conveniently into appdefs. Unfortunately in NeXus this does not exist, in OWL this could be possible but current NFDI activities go the opposite way they just create data schemas using OWL + the associated tools from the semantic web technology stack without ontology matching as the key priority and optimization aim

File: contributed_definitions/nyaml/NXcoordinate_system_set.yaml (hunk 3 above)
Comment (lukaspie) [reply contributed_definitions/nyaml/NXcoordinate_system_set.yaml:146]: How is this a different concept? You are not specifying anything in the docstring what makes this handedness special.. I guess it is "the handedness of the coordinate system processing_reference_frame" vs. "the handedness of any coordinate system" (as defined in the base class). However, for using them in an appdef, there is no difference, you would put there `processing_reference_frame/handedness` anyway, so it is clear you are talking about the handedness of processing_reference_frame, even if handedness is only defined in the bases class NXcoordinate_system.

File: contributed_definitions/nyaml/NXcoordinate_system_set.yaml (hunk 5 above)
Comment (mkuehbach) [reply contributed_definitions/nyaml/NXcoordinate_system_set.yaml:134]: Although my answer is tangible and to the point of NeXus' key weakness, a practical solution to go forward is indeed to move these specific coordinate system concepts (you on point here these are specific instances of concepts) but I do not want to repeat their writing in now NXapm, NXem, NXapm_PARAPROBE_TOOLS, NXapm_compositionspace_TOOLS, NXmicrostructure_CLASSES, mainly to avoid creating inconsistencies and having to copy over things which should be exactly the same, single source principles except them embedded in a different parent graphs' namespace.

File: contributed_definitions/nyaml/NXcoordinate_system_set.yaml (hunk 6 above)
Comment (mkuehbach) [reply contributed_definitions/nyaml/NXcoordinate_system_set.yaml:152]: Agreed

File: contributed_definitions/nyaml/NXcoordinate_system_set.yaml (hunk 4 above)
Comment (lukaspie) [reply contributed_definitions/nyaml/NXcoordinate_system_set.yaml:204]: Why can't you place it into `NXem `and then make `NXem_SOMETHING_MORE_SPECIFIC` extend NXem? This is the typical support application definition extension, we are using this also for `NXxps `(as extension of `NXmpes`). 

Then you put these coordinate systems in `NXem` once and `NXem_SOMETHING_MORE_SPECIFIC` just uses them without needing to copy them.

File: contributed_definitions/nyaml/NXcoordinate_system_set.yaml (hunk 7 above)
Comment (mkuehbach) [reply contributed_definitions/nyaml/NXcoordinate_system_set.yaml:168]: Exactly so then only possible in appdef but what if 20 appdefs agree on that aspect while showing differences on other parts of their concept tree, this is what I do not find convincing about NeXus, introducing reusable base classes that are allowed to make more domain specific conventions PROVIDED they otherwise build rigorously on core NIAC accepted base classes is in my opinion they way to go and maybe eventually even a practical way to go. I see that members in the NIAC may not feel to have the authority to decide what a specific community would like to agree on concept-wise, exactly this is exactly @yayahjb point about that also NeXus needs to face the reality of how to place oneself and be specific about what it serves it customers in a sea of options like UML, semantic web technology stack solutions and classical ad hoc XML or JSON-LD schema writing approaches that is a key question we have to ask ourselves which solution we propose.

My proposal is as I said:
1.) Core generic base classes (not a glossary of all human mankind terms)
2.) The man in the middle reusable concept sub-graphs (eventually allowing setting of constraints)
2.) Appdefs for the most specific combination of concepts + their constraints

File: contributed_definitions/nyaml/NXcoordinate_system_set.yaml (hunk 8 above)
Comment (mkuehbach) [reply contributed_definitions/nyaml/NXcoordinate_system_set.yaml:279]: Implemented

File: contributed_definitions/nyaml/NXcoordinate_system_set.yaml (hunk 9 above)
Comment (mkuehbach) [reply contributed_definitions/nyaml/NXcoordinate_system_set.yaml:313]: Correct, implemented, thank you very much for your on point comments

File: contributed_definitions/nyaml/NXcoordinate_system_set.yaml (hunk 10 above)
Comment (mkuehbach) [reply contributed_definitions/nyaml/NXcoordinate_system_set.yaml:346]: Okay so maybe indeed dropping the base class candidate NXconventions_em_ebsd was a bad choice or indeed copying it over into the NXem_ebsd base class.

BTW: NXem_ebsd is also one such discussion, strictly speaking the NIAC should not vote to support it if they stick to their argumentation of that base classes should be generic, but then all these required definitions to make in - just only one technique of dozens in EM already - would have to go all the way to the appdef. And that is always my point. You happen to agree with everybody about 99.9% the content and definitions in an appdef, but if one constraints you happen to be unable to agree upon you need to define a new appdef  and this has never been convincing to me and I never found anybody who attempted convincing why it is for them. Almost all of our proposed appdefs are on part in terms of thought and complexity covered to many existent appdefs in NeXus

File: contributed_definitions/nyaml/NXcoordinate_system_set.yaml (hunk 3 above)
Comment (mkuehbach) [reply contributed_definitions/nyaml/NXcoordinate_system_set.yaml:146]: You are right this is a point I think I forgot, processing_reference_frame inherits from NXcoordinate_system so handedness, i_alias, and i_direction fields can be removed UNLESS they specialize that concept via the docstring

File: contributed_definitions/nyaml/NXcoordinate_system_set.yaml (hunk 5 above)
Comment (lukaspie) [reply contributed_definitions/nyaml/NXcoordinate_system_set.yaml:134]: Ok, now I think I got your main point, namely that you want to have a place for conventions that are used across many (but of course not all) different domains, but you don't want to have them in the appdefs because you would need to repeat them a lot of times. Got it.

My suggestion for compromise would be the following: keep the specific NXcoordinate_systems here, but remove all of the unneeded repititions from the base class `NXcoordinate_system` (like handedness, and so on), and then remove anything that is strictly speaking domain-specific (like mentioning of a prefered direction, an electron beam, and so on). Then we can see if it is worth it to keeps these explicitly named coordinate systems (so they can be reused in em, apm, ...), without polluting the base class too much.

Comment (mkuehbach) [reply contributed_definitions/nyaml/NXcoordinate_system_set.yaml]: Exactly this was my point. Indeed, speculatively I followed your suggestion and pushed it as cc18a27b

File: contributed_definitions/nyaml/NXcoordinate_system_set.yaml [hunk 11]
```diff
@@ -88,138 +88,174 @@ doc: |
...
+      (e.g. rolling, transverse, and normal direction) align how with the direction of the base
+      vectors of the sample_reference_frame.
+      
+      Reference DOI: 10.1016/j.matchar.2016.04.008 suggest to label the
+      base vectors of this coordinate system as Xp, Yp, Zp.
+    origin(NX_CHAR):  # specialized the inherited concept origin from NXcoordinate_system
+      doc: |
+        Location of the origin of the processing_reference_frame.
+        
+        We assume regions-of-interest in this reference frame form a
```
Comment (lukaspie) [inline contributed_definitions/nyaml/NXcoordinate_system_set.yaml:151]: I would also drop the "we".

How about "It is assumed ..."

File: contributed_definitions/nyaml/NXcoordinate_system_set.yaml [hunk 12]
```diff
@@ -88,138 +88,174 @@ doc: |
...
+      doc: |
+        Direction of the positively pointing x-axis base vector of the processing_reference_frame.
+        We assume the configuration is inspected by looking towards the sample surface.
+        If a detector is involved, we assume the configuration is inspected from a position
+        that is located behind that detector.
+      enumeration: [undefined, north, east, south, west, in, out]
+    y_direction(NX_CHAR):
+      doc: |
+        Direction of the positively pointing y-axis base vector of the processing_reference_frame.
+        We assume the configuration is inspected by looking towards the sample surface.
```
Comment (lukaspie) [inline contributed_definitions/nyaml/NXcoordinate_system_set.yaml:167]: I think these assumptions are okay, but I would still allow users to define their own assumptions.

How about
```yaml
It is assumed the configuration is inspected by looking towards the sample surface. If a detector is involved, it is assumed the configuration is inspected from a position that is located behind that detector.

If any of these assumptions is not met, the user is asked to explicitly state this.
```
And then simply add these kinds of sentences everywhere where there is an assumption taken.

Alternatively, you could add such a statement it in the doc of the class itself, but I would prefer to have it with the individual concepts.

File: contributed_definitions/nyaml/NXcoordinate_system_set.yaml [hunk 13]
```diff
@@ -88,138 +88,174 @@ doc: |
...
+      doc: |
+        Direction of the positively pointing x-axis base vector of the sample_reference_frame.
+        We assume the configuration is inspected by looking towards the sample surface.
+        If a detector is involved, we assume the configuration is inspected from a position
+        that is located behind that detector.
+      enumeration: [undefined, north, east, south, west, in, out]
+    y_direction(NX_CHAR):
+      doc: |
+        Direction of the positively pointing y-axis base vector of the sample_reference_frame.
+        We assume the configuration is inspected by looking towards the sample surface.
```
Comment (lukaspie) [inline contributed_definitions/nyaml/NXcoordinate_system_set.yaml:211]: These repetitive statements could also just go to the docstring of sample_reference_frame, with a note that other assumptions are allowed (see above).

**Review by lukaspie — COMMENTED** [review]
This looks much cleaner already. I think you could clean up even more by stating the assumptions directly in the docs of the respective classes and adding a sentence that people who use different assumptions must explicitly state these.

File: contributed_definitions/nyaml/NXcoordinate_system_set.yaml (hunk 11 above)
Comment (mkuehbach) [reply contributed_definitions/nyaml/NXcoordinate_system_set.yaml:151]: Implemented

File: contributed_definitions/nyaml/NXcoordinate_system_set.yaml (hunk 12 above)
Comment (mkuehbach) [reply contributed_definitions/nyaml/NXcoordinate_system_set.yaml:167]: Agree that it should be with the concepts individually, enabling additional assumptions I support as the base class should offer service beyond materials engineering/condensed matter physics but then one cannot ask the user needs to make this explicit by overwriting the docstring of the concept, i.e.

I implemented:

If any of these assumptions is not met, the user is required to explicitly state this.

File: contributed_definitions/nyaml/NXcoordinate_system_set.yaml (hunk 13 above)
Comment (mkuehbach) [reply contributed_definitions/nyaml/NXcoordinate_system_set.yaml:211]: Implemented following the suggestion

File: contributed_definitions/nyaml/NXcoordinate_system_set.yaml (hunk 10 above)
Comment (mkuehbach) [reply contributed_definitions/nyaml/NXcoordinate_system_set.yaml]: Implemented

**Comment by mkuehbach** [comment]
All issues from @lukaspie were addressed.

**Comment by mkuehbach** [comment]
Neither conflicts nor revision requests currently block this PR from becoming merged into fairmat. However, as it was discussed for #270 development should continue despite vacation.

Therefore, this draft PR will be closed and instead sprint22_microstructure merged into sprint23_em_v3.
This enables to address merge conflicts between these two feature development activities to compose
one update for the EM line of features in NOMAD without currently breaking code without taking the
team the possibility to make revisions.
